products.php

```diff
@@ -0,0 +1,28 @@
+<?php
+header('Content-Type: application/json; charset=utf-8');
+header('Access-Control-Allow-Origin: *');
+
+require('vendor/autoload.php');
+
+use Furniture\Factories\DbConnector;
+use Furniture\Hydrators\ProductHydrator;
+use Furniture\Services\ResponseService;
+use \Furniture\Exceptions\InvalidCategoryException;
+
+const SUCCESS_MESSAGE = "Successfully retrieved products";
+
+try {
+    if (!is_numeric($_GET['cat'])) {
+        throw new InvalidCategoryException('INVALID CATEGORY ID');
+    }
+    $db = DbConnector::getDbConnection();
+    $products = ProductHydrator::fetchProductsByCategoryId($db, $_GET['cat']);
+    if (empty($products)) {
+        throw new Exception('No products found in database');
```

Comment:
If a category exists, but there's no inventory for it, it would probably be useful to display this information on the client, so maybe a special exception or response for this that the client can understand could be useful.

Reply:
discussed in-person